### ISSUE TITLE
Testing French/Multilang support for IMAGE in preparation for production.

### ISSUE THREAD

**Issue by notkaramel** [Shared-Reality-Lab/IMAGE-server#753]
# Testing Plan 
- Goal: to thoroughly test functionality of French support & validate the accuracy of the translation, all before uploading these changes to production (pegasus) server.

## Requirement:
- **Production** extension to version 0.4.48
- Customized server URL: against `unicorn`
  - https://unicorn.cim.mcgill.ca/image/

## Graphics to test:
- 4 types of graphics: photo; highcharts - line AND pie, embedded maps.
- I encourage testers to use graphics of their preference source to test for edge cases. 
- Example for Charts:
  - https://etherscan.io/charts
  - https://www.inspq.qc.ca/covid-19/donnees/age-sexe
- Random wikipedia page: https://wikiroulette.co 
- Available examples: https://image.a11y.mcgill.ca/pages/tutorial.html

## Tests
- All tests should be tested in all language settings (English, French, Automatic)
- For Linux users (and maybe MacOS?): 
  - Open a terminal session and enter the following command
  ```bash
  echo $LANGUAGE - $LANG
  ```
  - The output is your system's language and locale, separated by a hyphen
  - In case the output only show locale (`language` is blank), you can temporary set your locale to French then open your browser using the terminal
  ```bash
  export LANGUAGE=fr
  chromium
  ```

- [x] **TEST 1** Automatic English
  - Report your OS + browser's UI language. Make sure that IMAGE is displaying in English
  - Set language to "`Automatic`"
  - Send requests on all 4 types of graphics
  - Report the output (Do they look normal? Is there anything unusual happened?)
- [x] **TEST 2**: Automatic French
  - Report your OS + browser's UI language
  - Add French (fr, any variant) to your browser's "Preferred Language"
  - Set French to be the preferred language by set it to be the top language. Optionally can change the browser's display language (it will probably make the extension UI into French).
  - Set Language/Langue option to "Automatique"
  - Send requests of all types of graphics
  - Report the output (Do they look normal? Is there anything unusual happened?)
- [x] **TEST 3** Pour les francophones: French translation quality
  - Add and set French to be the top language of the browser. Make sure the extension UI is in French.
  - Evaluate the translation I made on the option page.
  - Set the language selection to 'Français'.
  - Send any type of graphics. Evaluate the translation display on the processing bar and result page.
  - Click on "Formulaire d'évaulation" and evaluate the translation there.
  - Report unclear translation & suggest any changes I should make to the text.

**Comment by notkaramel** [comment]
Test 1: On Windows, Microsoft Edge. UI is in English.
I spotted the option when right-click on a photo is still in French
![image](https://github.com/Shared-Reality-Lab/IMAGE-server/assets/24505220/e5b5869d-9a69-409f-a38f-e9d605be706a)

**Comment by jeffbl** [comment]
Notes on test procedure:
- Don't only pick graphics from the IMAGE website - try sites that you don't think other team member will be testing. I usually hit cnn.com, wsj.com, and/or washingtonpost.com. Sometimes a new article on wikipedia.
- 4 types of graphics? What other than photos, maps, charts?

**Comment by notkaramel** [comment]
> * Don't only pick graphics from the IMAGE website - try sites that you don't think other team member will be testing. I usually hit cnn.com, wsj.com, and/or washingtonpost.com. Sometimes a new article on wikipedia.

I edited the description to encourage testing on different source of graphics

> * 4 types of graphics? What other than photos, maps, charts?

I splitted charts into 2 types: Pie chart and Line chart, since they might differ in results. Line chart requires translation for one intepretation sentence, whereas Pie chart requires translation for each segments.

**Comment by notkaramel** [comment]
Inconsistency found in the browser extension: entry/sub-entry in the popup (context menu) is set to the language of the browser when the extension is downloaded. Will create a work item to fix this issue.

**Comment by notkaramel** [comment]
- I did some tests (1 and 2) here and there, seems like things are working fine. Couldn't replicate #755 with other graphics, seems to be an edge case.
- Charts and maps work fine, although some charts have renderings containing number with a parenthesis (`(15`) is not filtered by French-TTS regex which is a bit odd. Will look into that
- I'm still working on test 3 to validate the translation I made a while ago. Everything makes sense to me as an intermediate French speaker, but I think there might be some word that needs   better translation.
